shapely/tests/test_creation.py

```diff
@@ -58,9 +58,63 @@ def test_points_invalid_ndim():
 
 
 @pytest.mark.skipif(shapely.geos_version < (3, 10, 0), reason="GEOS < 3.10")
-def test_points_nan_becomes_empty():
+def test_points_nan_all_nan_becomes_empty():
     actual = shapely.points(np.nan, np.nan)
-    assert_geometries_equal(actual, shapely.Point())
+    assert actual.wkt == "POINT EMPTY"
+
+
+@pytest.mark.skipif(shapely.geos_version < (3, 12, 0), reason="GEOS < 3.12")
```

Comment:
Yes that is because of the behaviour with 3D empty points which differs in 3.11

I added some specific GEOS 3.11 tests and also notes in the docstring of the `points()` function.